pkg/api/v1/types.go

```diff
@@ -532,12 +532,21 @@ const (
 	ClaimLost PersistentVolumeClaimPhase = "Lost"
 )
 
+type PropagationMode string
+
+const (
+	PropagationPrivate PropagationMode = "private"
+	PropagationShared  PropagationMode = "shared"
+)
+
 // Represents a host path mapped into a pod.
 // Host path volumes do not support ownership management or SELinux relabeling.
 type HostPathVolumeSource struct {
 	// Path of the directory on the host.
 	// More info: http://releases.k8s.io/HEAD/docs/user-guide/volumes.md#hostpath
 	Path string `json:"path" protobuf:"bytes,1,opt,name=path"`
+	// Mount the host path with propagation mode specified. Docker only.
```

Comment:
The "Docker only" here means this PR currently implemented this feature in Docker runtime. The feature itself is not "Docker only".

see: https://github.com/opencontainers/runc/blob/master/libcontainer/specconv/spec_linux.go#L32

The word "PRIVATE", "SHARED", "SLAVE" are NOT invented by docker, but from linux kernel. As long as we are talking about linux container runtime, these modes should be inherited. @thockin do you feel better reuse current concept given it is introduced by linux?